pkg/infrastructure/clusterapi/ignition.go

```diff
@@ -202,7 +205,25 @@ func updateMCSCertKey(in IgnitionInput, platform string, rootCA *tls.RootCA, pri
 				config.Storage.Files[i].Contents.Source = &encoded
 
 				logrus.Debugf("Updated MCSCertKey file %s with new cert and key", mcsCertKeyFilepath)
-				break
+			case mcsKeyFile:
+				mcsKeyContents, err := yaml.Marshal(keyRaw)
+				if err != nil {
+					return fmt.Errorf("failed to marshal MCS Key: %w", err)
+				}
+				encoded := fmt.Sprintf("%s%s", replaceable, base64.StdEncoding.EncodeToString(mcsKeyContents))
+				// replace the contents with the edited information
+				config.Storage.Files[i].Contents.Source = &encoded
+				logrus.Debugf("Updated MCSKey file %s with new key", mcsKeyFile)
+
+			case mcsCertFile:
+				mcsCertContents, err := yaml.Marshal(certRaw)
+				if err != nil {
+					return fmt.Errorf("failed to marshal MCS Cert: %w", err)
+				}
+				encoded := fmt.Sprintf("%s%s", replaceable, base64.StdEncoding.EncodeToString(mcsCertContents))
```

Comment:
nit: I wonder if we should make this into a function. Or better yet, figure out why the ignition encoding gets the charset but simple encoding does not I do not think this should hold up the Pr though.